jobbergate-api/jobbergate_api/apps/job_submissions/schemas.py

```diff
@@ -274,6 +274,25 @@ class JobProperties(BaseModel, extra=Extra.forbid):
     )
     wckey: Optional[str] = Field(description="Specify wckey to be used with job.")
 
+    @validator("signal", pre=True)
+    def _backward_compatibility_on_signal(cls, value):
+        """
+        Remove the prefix B from signal in order to provide compatibility to legacy jobbergate applications.
+
+        The --signal parameter in Slurm commands allows for the specification of a signal to be sent to a job
+        in case of certain events, in order to trigger a specific behavior. The B syntax in sbatch specifies
+        a signal to be sent only to the job's batch shell, but is not supported by srun or the Slurm rest API,
+        since there is no batch shell to be signalized at runtime.
+
+        Notice the use the R option is possible to allow this job to overlap with a reservation with
+        MaxStartDelay set, so it should be preserved on the original value.
+
+        See test_backward_compatibility_on_signal_parameter for usage examples.
+        """
+        if isinstance(value, str) and ":" in value and "B" in value.split(":", 1)[0]:
+            return value.replace("B", "", 1).lstrip(":")
```

Comment:
I think this could be better implemented with a strict regular expression substitution.

Reply:
solved in [150e60d](https://github.com/omnivector-solutions/jobbergate/pull/262/commits/150e60d8549f54f041e3cc86f9425d104c1ccf3d)